pcluster/pcluster_stack.py

```diff
@@ -192,11 +192,23 @@ def __init__(self, scope: cdk.Construct, id: str, **kwargs) -> None:
         poweruser_group.add_managed_policy(iam.ManagedPolicy.from_aws_managed_policy_name('AWSCloud9Administrator'))
 
         # HPC User
-        user = iam.User(self, 'Researcher',
-                        password=cdk.SecretValue.cfn_parameter(password), password_reset_required=True)
-        user.add_to_group(group=admin_group)
-        cdk.CfnOutput(self, 'UserLoginUrl', value="".join(["https://", self.account,".signin.aws.amazon.com/console"]))
-        cdk.CfnOutput(self, 'UserName', value=user.user_name)
+        user = iam.CfnUser(self, 'Researcher',
+                           groups=[admin_group.node.default_child.ref],
+                           login_profile=iam.CfnUser.LoginProfileProperty(
+                                   password_reset_required=True,
+                                   password=cdk.SecretValue.cfn_parameter(password).to_string()
+                               )
+                           )
+
+        create_user = cdk.CfnParameter(self, "CreateUser", default="false", type="String", allowed_values=['true','false']).value_as_string
+        user.cfn_options.condition = cdk.CfnCondition(self, "UserCondition", expression=cdk.Fn.condition_equals(create_user, "true"))
+
+
+        output_user_loginurl = cdk.CfnOutput(self, 'UserLoginUrl', value="".join(["https://", self.account,".signin.aws.amazon.com/console"]))
+        output_user_loginurl.condition = cdk.CfnCondition(self, "OutputUserLoginUrlCondition", expression=cdk.Fn.condition_equals(create_user, "true"))
```

Comment:
Can we re-use the same condition created above for the user?

Reply:
Good point. Simplified.